Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -128,6 +128,20 @@ - (void)_update {
     self.updating = YES;
     NSAssert([NSThread isMainThread], @"Update must be called on the main thread");
     NSManagedObjectContext *moc = self.dataStore.viewContext;
+    
+    NSFetchRequest *notSavedToDiskRequest = [WMFArticle fetchRequest];
+    notSavedToDiskRequest.predicate = [NSPredicate predicateWithFormat:@"savedDate != NULL && isSavedToDisk != YES"];
+    notSavedToDiskRequest.fetchLimit = 1;
+    NSError *notSavedToDiskError = nil;
+    WMFArticle *articleToSaveToDisk = [[self.dataStore.viewContext executeFetchRequest:notSavedToDiskRequest error:&notSavedToDiskError] firstObject];
+    if (notSavedToDiskError) {
+        DDLogError(@"Error fetching article to save to disk: %@", notSavedToDiskError);
+    }
+    if (articleToSaveToDisk.URL) {
```

Comment:
Looks like this `if` only needs to happen if `notSavedToDiskError` is nil.

Reply:
Wouldn't `articleToSaveToDisk` be nil if fetch failed? If you look at line 150 and 159, we're not checking if `fetchError` is `nil`, same on line 182 and 185, not checking if `downloadedFetchError` is `nil`?

Reply:
Yeah this whole fetcher beast is a bit of a Frankensteins monster :) If you're down it would be cool if L132-L143 were just pulled out into a method, then you can just return in the error case after DDLog.

Reply:
(ala Swifty `guard` pattern)